java/test/src/main/java/io/ray/test/ObjectStoreTest.java

```diff
@@ -37,4 +69,26 @@ public void testGetMultipleObjects() {
     List<ObjectRef<Integer>> refs = ints.stream().map(Ray::put).collect(Collectors.toList());
     Assert.assertEquals(ints, Ray.get(refs));
   }
+
+  @Test
+  public void testPutWithAssignedOwner() throws Exception {
+    try {
+      ActorHandle<Creator> creator = Ray.actor(Creator::new).remote();
+      ActorHandle<Owner> owner = Ray.actor(Owner::new).remote();
+      ActorHandle<Borrower> borrower = Ray.actor(Borrower::new).remote();
+      Ray.get(owner.task(Owner::warmup).remote());
+      ObjectRef<ObjectRef<Integer>> ref = creator.task(Creator::put, 1, owner).remote();
+      Ray.get(owner.task(Owner::set, ref).remote());
+      creator.task(Creator::exit).remote();
+      Thread.sleep(10000);
+      int data = Ray.get(borrower.task(Borrower::get, ref).remote());
+      Assert.assertEquals(data, 1);
+    } catch (RayTaskException e) {
```

Comment:
I think you missed the final part of the corresponding Python test:

https://github.com/ray-project/ray/blob/492076806d2f691fc43d35016cba857a2235d0b3/python/ray/tests/test_object_assign_owner.py#L73-L77